src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -377,12 +382,33 @@ protected void addAdditionalContextValues(
         String extensionUniqueId = getThreadContext().getHeader("extension_unique_id");
         if (extensionUniqueId != null) {
             ExtensionsManager extManager = OpenSearchSecurityPlugin.GuiceHolder.getExtensionsManager();
-            if (extManager.lookupExtensionSettingsById(extensionUniqueId).isPresent()) {
+            Optional<ExtensionsSettings.Extension> extension = extManager.lookupExtensionSettingsById(extensionUniqueId);
+
+            if (extension.isPresent()
+                && isExtensionAllowed((List<String>) extension.get().getAdditionalSettings().get(EXTENSION_NODES_DN), principal)) {
                 getThreadContext().putTransient(ConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_EXTENSION_REQUEST, Boolean.TRUE);
             }
         }
         // CS-ENFORCE-SINGLE
 
         super.addAdditionalContextValues(action, request, localCerts, peerCerts, principal);
     }
+
+    public boolean isExtensionAllowed(List<String> extensionDistinguishedNames, String principal) {
```

Comment:
Is it possible to make this `static`? If this is static I think it could remove the need to instantiate a SecurityRequestHandler in the test file to test this method.

This PR looks good to me!

Reply:
Agreed, changed to static